File: glassshop-backend/models/QuotationItem.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const QuotationItem = sequelize.define('QuotationItem', {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true
    },
    quotationId: {
      type: DataTypes.BIGINT,
      allowNull: false,
      field: 'quotation_id',
      references: {
        model: 'quotations',
        key: 'id'
      }
    },
    glassType: {
      type: DataTypes.STRING(100),
      allowNull: false,
      field: 'glass_type'
    },
    thickness: {
      type: DataTypes.STRING(50),
      allowNull: true
    },
    height: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false
    },
    width: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false
    },
    heightUnit: {
      type: DataTypes.STRING(10),
      allowNull: true,
      defaultValue: 'FEET',
      field: 'height_unit'
    },
    widthUnit: {
      type: DataTypes.STRING(10),
      allowNull: true,
      defaultValue: 'FEET',
      field: 'width_unit'
    },
    design: {
      type: DataTypes.STRING(50),
      allowNull: true
    },
    quantity: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 1
    },
    ratePerSqft: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
      field: 'rate_per_sqft'
    },
    area: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false
    },
    subtotal: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false
    },
    hsnCode: {
      type: DataTypes.STRING(10),
      allowNull: true,
      field: 'hsn_code'
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    itemOrder: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
      field: 'item_order'
    },
    createdAt: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
      field: 'created_at'
    }
  }, {
    tableName: 'quotation_items',
    timestamps: false,
    underscored: true
  });

  return QuotationItem;
};

File: GlassShop/src/main/java/com/glassshop/ai/entity/QuotationItem.java
package com.glassshop.ai.entity;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "quotation_items")
public class QuotationItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "quotation_id", nullable = false)
    private Quotation quotation;

    @Column(name = "glass_type", nullable = false, length = 100)
    private String glassType;

    @Column(name = "thickness", length = 50)
    private String thickness;

    @Column(name = "height", nullable = false)
    private Double height;

    @Column(name = "width", nullable = false)
    private Double width;

    @Column(name = "height_unit", length = 10)
    private String heightUnit = "FEET";

    @Column(name = "width_unit", length = 10)
    private String widthUnit = "FEET";

    @Column(name = "design", length = 50)
    private String design;

    @Column(name = "quantity", nullable = false)
    private Integer quantity = 1;

    @Column(name = "rate_per_sqft", nullable = false)
    private Double ratePerSqft;

    @Column(name = "area", nullable = false)
    private Double area;

    @Column(name = "subtotal", nullable = false)
    private Double subtotal;

    @Column(name = "hsn_code", length = 10)
    private String hsnCode;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "item_order", nullable = false)
    private Integer itemOrder = 0;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Quotation getQuotation() {
        return quotation;
    }

    public void setQuotation(Quotation quotation) {
        this.quotation = quotation;
    }

    public String getGlassType() {
        return glassType;
    }

    public void setGlassType(String glassType) {
        this.glassType = glassType;
    }

    public String getThickness() {
        return thickness;
    }

    public void setThickness(String thickness) {
        this.thickness = thickness;
    }

    public Double getHeight() {
        return height;
    }

    public void setHeight(Double height) {
        this.height = height;
    }

    public Double getWidth() {
        return width;
    }

    public void setWidth(Double width) {
        this.width = width;
    }

    public String getHeightUnit() {
        return heightUnit;
    }

    public void setHeightUnit(String heightUnit) {
        this.heightUnit = heightUnit;
    }

    public String getWidthUnit() {
        return widthUnit;
    }

    public void setWidthUnit(String widthUnit) {
        this.widthUnit = widthUnit;
    }

    public String getDesign() {
        return design;
    }

    public void setDesign(String design) {
        this.design = design;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public Double getRatePerSqft() {
        return ratePerSqft;
    }

    public void setRatePerSqft(Double ratePerSqft) {
        this.ratePerSqft = ratePerSqft;
    }

    public Double getArea() {
        return area;
    }

    public void setArea(Double area) {
        this.area = area;
    }

    public Double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(Double subtotal) {
        this.subtotal = subtotal;
    }

    public String getHsnCode() {
        return hsnCode;
    }

    public void setHsnCode(String hsnCode) {
        this.hsnCode = hsnCode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getItemOrder() {
        return itemOrder;
    }

    public void setItemOrder(Integer itemOrder) {
        this.itemOrder = itemOrder;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

